File: 2022/kotlin/kotlin/src/main/kotlin/aoc/days/day01/day.kt
package aoc.days.day01

import aoc.days.day0.DayBaseClass
import aoc.utils.ReadInput

class Day (testInput: Boolean = false): DayBaseClass(testInput) {

    @Override
    override fun readInput(){
        this.input = ReadInput.toString(this.path)
    }

    @Override
    override fun solve() {
        val sums  = (this.input as String).split("\r\n\r\n")
            .map { it.split("\r\n").sumOf { it.toInt() }}.toMutableList()
        sums.sort()
        this.p1 = sums.takeLast(1).sum()
        this.p2 = sums.takeLast(3).sum()
    }
}
fun main(){
    Day()
}

File: 2022/js/DAY01/day.js
let readfile = require('../utils/readfile');

let data = readfile({ split: false })

// data.forEach((e) => {
//     console.log(e);
// })

console.log()
let sums = data.split('\n\n')
    .map((e) => { return e.split('\n').map((f) => { return parseInt(f) }).reduce((a, b) => a + b) }).sort((a, b) => (a - b))
console.log("P1", sums.slice(-1).reduce((a, b) => a + b))
console.log("P2", sums.slice(-3).reduce((a, b) => a + b))

// console.log(sums)